Source/WebCore/Modules/webaudio/AudioWorkletGlobalScope.cpp

```diff
@@ -117,7 +120,11 @@ ExceptionOr<void> AudioWorkletGlobalScope::registerProcessor(String&& name, Ref<
         }
     }
 
-    auto addResult = m_processorConstructorMap.add(name, WTFMove(processorConstructor));
+    decltype(m_processorConstructorMap)::AddResult addResult;
+    {
+        Locker locker { m_processorConstructorMapLock };
+        addResult = m_processorConstructorMap.add(name, WTFMove(processorConstructor));
```

Comment:
I recommend we use `bool isNewEntry` variable instead of `addResult`. `addResult` contains an iterator which would be unsafe to use outside the scope of the lock, so it's a foot gun.

Reply:
Great point. Given your other comment though we can probably avoid all this locking if I get the worklet's context added as an opaque root instead.